tool/tsh/tshconfig.go

```diff
@@ -46,6 +52,26 @@ type ExtraProxyHeaders struct {
 	Headers map[string]string `yaml:"headers,omitempty"`
 }
 
+// Merge two configs into one. The config from parameter has higher priority.
+func (config *TshConfig) Merge(otherConfig *TshConfig) TshConfig {
+	baseConfig := config
+	if baseConfig == nil {
```

Comment:
Why is the `nil` check necessary here? The `loadAllConfigs` is not ignoring errors from `loadConfig` (which could generate those `nil` pointers).

In addition, it feels a bit contra-intuitive to call a method from a `nil` pointer (even if it is possible), so calling something like `(*config)(nil).Merge(otherConfig)` would return a valid result.

Reply:
Semantically `nil` and `&TshConfig{}` has exactly the same amount of information: zero. These assignments make it easier to implement the rest of the function. In the future (when I implement the rest of the RFD) there will be more fields to merge and having nil pointers makes it uglier.